ls8/cpu.py

```diff
@@ -62,4 +99,64 @@ def trace(self):
 
     def run(self):
         """Run the CPU."""
-        pass
+        # load the program
+        self.load()
+
+        while True:
+            # Instruction Register, contains a copy of the currently executing instruction
+            IR = self.ram_read(self.pc)
+            # Grab AA of the program instruction for the operand count
+            # Grab C of the program instruction for if the instruction sets PC counter
+            operand_count = IR >> 6
+            sets_pc = IR >> 4 & 0b0001
+
+            if IR == LDI:
+                address = self.ram_read(self.pc + 1)
+                value = self.ram_read(self.pc + 2)
+                # store the data
+                self.reg[address] = value
+                # increment the PC by 3 to skip the arguments
+            elif IR == PRN:
+                data = self.ram_read(self.pc + 1)
+                # print the data
+                print(self.reg[data])
+                # increment the PC by 2 to skip the argument
+            elif IR == MUL or IR == ADD:
+                op = IR
+                reg_a = self.ram_read(self.pc + 1)
+                reg_b = self.ram_read(self.pc + 2)
+                self.alu(op, reg_a, reg_b)
```

Comment:
Maybe the assignment to `op` isn't needed, `self.alu(IR, reg_a, reg_b)`, but I get why you would remind yourself that it's the current operation